test/functional/rpc_generateblock.py

```diff
@@ -51,37 +51,34 @@ def run_test(self):
         assert_equal(len(block['tx']), 1)
         assert_equal(block['tx'][0]['vout'][0]['scriptPubKey']['addresses'][0], combo_address)
 
-        # Generate 110 blocks to spend
-        node.generatetoaddress(110, address)
+        # Generate some blocks to spend
+        miniwallet.generate(200)
 
         # Generate some extra mempool transactions to verify they don't get mined
         for _ in range(10):
-            node.sendtoaddress(address, 0.001)
+            miniwallet.send_self_transfer(from_node=node)
 
         self.log.info('Generate block with txid')
-        txid = node.sendtoaddress(address, 1)
+        txid = miniwallet.send_self_transfer(from_node=node)['txid']
         hash = node.generateblock(address, [txid])['hash']
         block = node.getblock(hash, 1)
         assert_equal(len(block['tx']), 2)
         assert_equal(block['tx'][1], txid)
 
         self.log.info('Generate block with raw tx')
-        utxos = node.listunspent(addresses=[address])
-        raw = node.createrawtransaction([{'txid':utxos[0]['txid'], 'vout':utxos[0]['vout']}],[{address:1}])
-        signed_raw = node.signrawtransactionwithwallet(raw)['hex']
-        hash = node.generateblock(address, [signed_raw])['hash']
+        txid = miniwallet.send_self_transfer(from_node=node)['txid']
+        rawtx = node.getrawtransaction(txid)
+        hash = node.generateblock(address, [rawtx])['hash']
         block = node.getblock(hash, 1)
         assert_equal(len(block['tx']), 2)
         txid = block['tx'][1]
-        assert_equal(node.gettransaction(txid)['hex'], signed_raw)
+        assert_equal(node.getrawtransaction(txid, False, hash), rawtx)
 
         self.log.info('Fail to generate block with out of order txs')
-        raw1 = node.createrawtransaction([{'txid':txid, 'vout':0}],[{address:0.9999}])
-        signed_raw1 = node.signrawtransactionwithwallet(raw1)['hex']
-        txid1 = node.sendrawtransaction(signed_raw1)
-        raw2 = node.createrawtransaction([{'txid':txid1, 'vout':0}],[{address:0.999}])
-        signed_raw2 = node.signrawtransactionwithwallet(raw2)['hex']
-        assert_raises_rpc_error(-25, 'TestBlockValidity failed: bad-txns-inputs-missingorspent', node.generateblock, address, [signed_raw2, txid1])
```

Comment:
It doesn't seem totally related to what's being tested here, but one thing I noticed while reviewing:

Do you know why, in the original code, `rpc_generateblock.py` sometimes does  `sendrawtransaction` and other times does not, before `generateblock`?

ie here `signed_raw2` was never submitted to the `node` with `sendrawtransaction`, while `txid1` was.

Reply:
I am not entirely sure, but my assumption while reading through the code was that it was to save a line of code.

Reply:
`sendrawtransaction` submits the tx to mempool, while `createrawtransaction` + `signrawtransaction` merely creates the raw tx objects. Here, in the original test, tx1 is submitted and tx2 is not. It's calling `generateblock()` with the whole raw tx for tx2 and just the txid for tx1.

So, actually, your implementation doesn't really match what was there before. Be careful - [`generateblock`](https://github.com/bitcoin/bitcoin/blob/417f95fa453d97087a33d4176523ab278bef21a1/src/rpc/mining.cpp#L295) accepts hexstrings that can represent either txids (when the tx is already in mempool) or raw transactions.

Reply:
I did notice that but if this test is testing for out of order transactions isn't the distinction between a raw transaction and a transaction id arbitrary?